tests/test_least_squares.py

```diff
@@ -0,0 +1,54 @@
+from lmfit import Parameters, minimize, fit_report, Minimizer
+from lmfit_testutils import assert_paramval, assert_paramattr
+
+from numpy import linspace, zeros, sin, exp, random, pi, sign
+
```

Comment:
Can we have a test for:

1. that fits with `Parameter.expr` still work?
2. to check that the output from `least_sq` is the same as `leastsq`, as far as we know it should be?
3. that error estimation is correct?